### DISCUSSION TITLE
Hand made maps

### DISCUSSION THREAD

**Discussion by MikeBravo15** [XCSoar/xcsoar-data-content#334] [General]
Hallo all,
is there a possibility to share hand-made maps in the xcsoar-data-content section? If yes, can you please describe how to? Thank you for your kind help

**Comment by lordfolken** [comment]
What would these maps contain, or do differently than the ones we have? If its just an area, we can add it to 
https://github.com/XCSoar/xcsoar-data-content/tree/master/data/source/map as a new map via the coordinates. 

In general i want maps that are reproducable. The primary problem at this point is that the osm topology we use is very old. 
(I'm currently working on this..) 

If you have layers or data that you think xcsoar's maps would benefit from, please do share. So it can be integrated.

**Comment by MikeBravo15** [comment]
Well it depens. I live in area, where OSM data are more than acceptable, but it do not have to be true for the whole world. I created hires and lores handmade maps for Czecho-Slovak region (I cannot enclose them here as file limit is 25MB, maps are 54 for lo-res elevation map/65MB for hi-res ). All the datas are actually OSM based with exception of city limits, which I was not able to obtain from OSM, so I used Sentinel-2 10m land cover time series of the world https://www.esri.com/partners/impact-observatory-a2T5x0000084pJXEAY/sentinel-2-10m-land--a2d5x000005jw9NAAQ. It also contains a new layer for forrests. Problem with this layer is size - for two relativelly small countries it is kinda acceptable to have this layer, but I still had to optimize shapes a lot, but for Germany/France/Poland I guess forrests are too big to get a decent performance. Therefore I believe, that handmade maps with more details for smaller areas are still a vitable option.
![Screenshot (14  5](https://github.com/XCSoar/xcsoar-data-content/assets/42399455/6d0ec2e6-8c0e-4240-b03c-a5229682d860)

**Reply by MikeBravo15** [reply]
Sentinel data can be obtained here

https://livingatlas.arcgis.com/landcoverexplorer/#mapCenter=14.801%2C50.009%2C9&mode=step&timeExtent=2017%2C2022&year=2022&downloadMode=true

**Reply by MikeBravo15** [reply]
.xcm files here:
https://ulozto.cz/tam/12bd1a2e-1cdf-49e2-aedc-762f8d549295

**Comment by lordfolken** [comment]
Thanks for the sentinel link. I was looking for something that would show the proper city layouts. 
In OSM i tried to filter for landuse, but its futile. And administrative boundary is treated differently across the globe. 

"Therefore I believe, that handmade maps with more details for smaller areas are still a vitable option."
The problem is maintainability. At some point someone is gonna come and say X is broken. (Like the current elbe ticket..)  
And then experience over the last 10 years shows that the person that built the thing is no where to be found. And then i will have to invest time in that one use case again. 

I'm fine with adding switches in for a forest layer or similar in the map source config. But manually built stuff is always legacy and of no real value.

**Comment by lordfolken** [comment]
I'm a bit confused by the landsat layer. This seems to be a geotiff dataset, yet you seems to have shapefiles from somewhere?

**Reply by MikeBravo15** [reply]
I had to convert it from raster to vector via a Polygonize tool (I used QGIS for this). QGIS is opensource - https://github.com/qgis/QGIS so how this tool works can be probably found there and used for processing the geotiff data

**Reply by lordfolken** [reply]
I use QGIS extensively. Just wasn't clear whether there was a direct data set.

**Reply by lordfolken** [reply]
https://github.com/lordfolken/xcsoar-mapgen-postgis i have a qgis style in there that should more or less look like an xcsoar map, in connection to a postgis server. 
Although the drawing distance configuration isn't yet in there.

**Reply by lordfolken** [reply]
I did also a lot of experimentation with ogr2ogr https://github.com/lordfolken/mapgen-datagen

**Comment by lordfolken** [comment]
I don´t know if you have seen this: I have built https://github.com/XCSoar/xcsoar-mapgen-topology which is able to extract various layers of osm data. It also reduces the shapes and polygon count, to a more manageable size. 

Inspired by you, i also extracted forests and landuse. However i haven't included them yet in XCSoar, as i find that xcsoar's polygon renderer and especially decluttering mechansims are not ideal.

**Reply by Sundown3867** [reply]
Thank you for your effort - during my tests I found that XCSoar do not draw holes in polygons. Forrest polygons exported from OSM had a guite a lot of inner polygons a also overlaps. Overlaps are drawn in darker color in XCS (alpha channel utilized)

**Reply by Sundown3867** [reply]
Hallo Philipp, I have recently converted city area from latest ESRI geotiffs to one global shape file. Filesize ia cca 1.9GB for the whole world for city areas bigger than 1sqare km. I run some optimalization on the file but still keep original data (a lot of smaller files for individual tiles) if needed. If you want use it for XCSoar map generator, let me know and I will upload it somewhere.

**Reply by lordfolken** [reply]
Id love to have it on mapgen-data.sigkill.ch. Do you have an ssh key?
